core/modules/user/user.module

```diff
@@ -2263,21 +2253,30 @@ function user_roles($members_only = FALSE, $permission = NULL, $full_objects = F
 }
 
 /**
- * Helper for user_roles().
+ * Checks if access should be restricted.
```

Comment:
How about "Checks a role to see if any of its permissions have the 'restrict access' flag set."